src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1667,6 +1693,49 @@ protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArgum
             responder.SetResponse(response);
         }
 
+        // test -- need to delete
+        protected override void HandleDisassembleRequestAsync(IRequestResponder<DisassembleArguments, DisassembleResponse> responder)
+        {
+            DisassembleResponse response = new DisassembleResponse();
+
+            int hr;
+            DisassembleArguments disassembleArguments = responder.Arguments;
+            Debug.Assert(!string.IsNullOrEmpty(disassembleArguments.MemoryReference));
+            try
+            {
+                IDebugMemoryContext2 memoryContext = GetMemoryContext(disassembleArguments.MemoryReference, disassembleArguments.Offset, out ulong address, out hr);
+                if (m_program.GetDisassemblyStream(enum_DISASSEMBLY_STREAM_SCOPE.DSS_ALL, (IDebugCodeContext2)memoryContext , out IDebugDisassemblyStream2 disassemblyStream) == HRConstants.S_OK)
+                {
+                    if (disassemblyStream.Seek(enum_SEEK_START.SEEK_START_BEGIN, (IDebugCodeContext2)memoryContext, address, (long)disassembleArguments.InstructionOffset) == HRConstants.S_OK)
+                    {
+                        DisassemblyData[] prgDisassembly = new DisassemblyData[disassembleArguments.InstructionCount];
+                        if (disassemblyStream.Read((uint)disassembleArguments.InstructionCount, enum_DISASSEMBLY_STREAM_FIELDS.DSF_ALL, out uint pdwInstructionsRead, prgDisassembly) == HRConstants.S_OK)
+                        {
+                            Debug.Assert(disassembleArguments.InstructionCount == pdwInstructionsRead);
+                            foreach (DisassemblyData data in prgDisassembly)
+                            {
+                                DisassembledInstruction instruction = new DisassembledInstruction() {
+                                    Address = data.bstrAddress,
+                                    InstructionBytes = data.bstrCodeBytes,
+                                    Instruction = data.bstrOpcode,
+                                    Symbol = data.bstrSymbol,
+                                    // Location = data.uCodeLocationId,
+                                    Line = (int?)data.posBeg.dwLine,
+                                    Column = (int?)data.posBeg.dwColumn,
+                                    EndLine = (int?)data.posEnd.dwLine,
+                                    EndColumn = (int?)data.posEnd.dwColumn
+                                };
+                                response.Instructions.Add(instruction);
+                            }
+                        }
+                    }
+
+                }
+            } catch { }
```

Comment:
>{ [](start = 20, length = 1)

You want a `responder.SetError` call in the catch and move `responder.SetResponse(response);` to the end of the try block.